include/spl/sys/procfs_list.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+/*
+ * Copyright (c) 2018 by Delphix. All rights reserved.
+ */
+
+#ifndef	_SPL_PROCFS_LIST_H
```

Comment:
If that's the case, one could make the argument that since this is Linux-specific code, it shouldn't go to the SPL because it is not something that comes from the Solaris codebase (e.g. there is no `procfs_list` in illumos).

In any case, the code in question is not very big in size so this doesn't seem like a big deal.

Reply:
The reality is the SPL is increasingly misnamed, but functionally it still acts as an abstraction between the _mostly_ portable ZFS code and the platform specific code.  It would be nice to eventually adopt a generic shared ZFS compatibility layer which provides interfaces which aren't wed to a single platform.